src/main/java/com/thealgorithms/datastructures/heaps/SortAccFrequency.java

```diff
@@ -0,0 +1,35 @@
+package com.thealgorithms.datastructures.heaps;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.PriorityQueue;
+
+public class SortAccFrequency {
+
+    static String frequencySort(String str) {
+        Map<Character, Integer> map = new HashMap<>();
+        for (char c : str.toCharArray()) {
+            map.put(c, map.getOrDefault(c, 0) + 1);
+        }
+        
+        for (char c : str.toCharArray()) {
+            System.out.println(c + " " + map.get(c));
+        }
+        
+        PriorityQueue<Character> maxHeap = new PriorityQueue<>((a, b) -> map.get(b) - map.get(a));
+        maxHeap.addAll(map.keySet());
+        
+        StringBuilder res = new StringBuilder();
+        while (!maxHeap.isEmpty()) {
+            char c = maxHeap.poll();
+            int freq = map.get(c);
+            for (int i = 0; i < freq; i++) {
+                res.append(c);
+            }
+        }
+        return res.toString();
+    }
+
+    public static void main(String[] args) {
+        System.out.println(frequencySort("aabbsssdd"));
+    }
```

Comment:
Please remove the `main` method and instead add some proper tests. Good examples to look at:
- [`Factorial`](https://github.com/TheAlgorithms/Java/blob/566c27a996f48a94916170c159ae735546b39c42/src/main/java/com/thealgorithms/maths/Factorial.java) and [`FactorialTest`](https://github.com/TheAlgorithms/Java/blob/566c27a996f48a94916170c159ae735546b39c42/src/test/java/com/thealgorithms/maths/FactorialTest.java),
- [`FindMax`](https://github.com/TheAlgorithms/Java/blob/566c27a996f48a94916170c159ae735546b39c42/src/main/java/com/thealgorithms/maths/FindMax.java) and [`FindMaxTest`](https://github.com/TheAlgorithms/Java/blob/566c27a996f48a94916170c159ae735546b39c42/src/test/java/com/thealgorithms/maths/FindMaxTest.java).